### ISSUE TITLE
--depnotify flag not working a reinstalled macOS

### ISSUE THREAD

**Issue by xlonil** [grahampugh/erase-install#165]
**Describe the bug**
While testing the `--depnotify` flag that was introduced in version 24.1 on a recently reinstalled macOS, I noticed that it fails the first run. Running erase-install a second time fixes the issue. This seems to happen if DEPNotify isn't already available at `/Applications/Utilities/DEPNotify.app`.

**To Reproduce**
Remove `/Applications/Utilities/DEPNotify.app`
Remove `/Library/Management` folder.
Run `./erase-install.sh --erase --depnotify` again.
 
**Expected behavior**
Expecting `erase-install.sh` to download DEPNotify - if not already installed.

**Code/log output**
```
./erase-install.sh --erase --overwrite --depnotify --check-power --test-run
   [get_depnotify] Downloading DEPNotify.app...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0Warning: Failed to create the file 
Warning: /Library/Management/erase-install/DEPNotify.pkg: No such file or 
Warning: directory
  0 5338k    0  2896    0     0  25403      0  0:03:35 --:--:--  0:03:35 25403
curl: (23) Failed writing body (0 != 2896)
   [get_depnotify] DEPNotify download failed
   [get_depnotify] Could not download DEPNotify.app.

   [erase-install] v24.1 script execution started: Mon Nov 22 11:26:39 CET 2021
   [erase-install] Caffeinating this script (pid=15635)

*** TEST-RUN ONLY! ***
* This script will perform all tasks up to the point of erase or reinstall,
* but will not actually erase or reinstall.
* Remove the --test-run argument to perform the erase or reinstall.
**********************

   [free_space_check] OK - 416010188 KB free disk space detected
   [check_power_status] OK - AC power detected
   [erase-install] Looking for existing installer app or pkg
   [find_existing_installer] No valid installer found.
   [erase-install] Opening osascript dialog (language=en)
   [get_installinstallmacos] Making working directory at /Library/Management/erase-install
   [get_installinstallmacos] Downloading installinstallmacos.py...
/Library/Management/erase-install/installinstallmacos.py: OK
   [get_installinstallmacos] downloaded new installinstallmacos.py successfully.
   [get_installinstallmacos] installinstallmacos.py is in /Library/Management/erase-install
   [get_relocatable_python] MacAdmins Python is installed
   [run_installinstallmacos] Getting current production version

   [run_installinstallmacos] This command is now being run:

   installinstallmacos.py --workdir /Library/Management/erase-install --ignore-cache --raw --auto
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0 11.5G    0 62.9M    0     0  17.7M      0  0:11:08  0:00:03  0:11:05 17.7M
```

**Environment (please complete the following information):**
 - OS version 11.6.1
 - erase-install version 24.1
 
**Summary**
Looking for clues in the `erase-install.sh` code (and the log above), it seems that the working directory `/Library/Management/erase-install` is not present prior to `get_installinstallmacos` function being run.

**Comment by grahampugh** [comment]
It looks like DEPNotify failed to download, which I would presume is a network issue:
```
   [get_depnotify] Downloading DEPNotify.app...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0Warning: Failed to create the file 
Warning: /Library/Management/erase-install/DEPNotify.pkg: No such file or 
```

If you have unreliable internet, you would be better to deploy the release package that I supply that includes DEPNotify (though of course, if you have unreliable internet, you might also struggle to get the macOS installer 😀 ).

**Comment by grahampugh** [comment]
It works for me. I think you need to check your network.

```
sudo /Library/Management/erase-install/erase-install.sh --test-run --no-fs --reinstall --current-user --check-power --depnotify
   [get_depnotify] Downloading DEPNotify.app...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 5338k  100 5338k    0     0  5137k      0  0:00:01  0:00:01 --:--:-- 5163k
installer: Package name is DEPNotify
installer: Upgrading at base path /
installer: The upgrade was successful.
   [get_depnotify] DEPNotify is installed
```

**Comment by xlonil** [comment]
I thought so at first, but this is tested on two different locations and network providers. One of them being the University network. Would that be the case, yes, we could always package DEPNotify by itself, but you had such a nice option to have it downloaded for us :)

Ah, I think I see whats going wrong here. This issue is only present if the folder `/Library/Management/erase-install/` does not exist. Since you have and are working from the `/Library/Management/erase-install/` folder, then it is going to work. We have `erase-install.sh` installed at a different location on our devices.

For what I can see, there is a `mkdir` which will create the working directory if not exist, but it is being run after DEPNotify tries to download. This would explain why it always fails the first time around. After being run a second time, the working directory has been created and everything runs smoothly.

```
get_installinstallmacos() {
    # grab installinstallmacos.py if not already there
    # note this does a SHA256 checksum check and will delete the file and exit if this fails
    if [[ ! -d "$workdir" ]]; then
        echo "   [get_installinstallmacos] Making working directory at $workdir"
        mkdir -p "$workdir"
    fi
```

**Comment by grahampugh** [comment]
Ah,

**Comment by grahampugh** [comment]
Should be fixed now in v25.0. Let me know if not.
